src/nspawn/nspawn.c

```diff
@@ -3380,6 +3380,9 @@ static int merge_settings(Settings *settings, const char *path) {
                 strv_free_and_replace(arg_parameters, settings->parameters);
         }
 
+        if ((arg_settings_mask & SETTING_EPHEMERAL) == 0)
+                arg_ephemeral = settings->ephemeral;
```

Comment:
for this to work correctly, when we parse --ephemeral from the command line we need to do:

```c
arg_settings_mask |= SETTING_EPHEMERAL;
```

(see how this is done for the other command line arguments, for example `arg_read_only`)

otherwise having an .nspawn file around would mean we always turn off ephemeral mode by default, and that's definitely not desired.